### ISSUE TITLE
Local SDK Available

### ISSUE THREAD

**Issue by rgerrans** [andrei-tatar/node-red-contrib-nora#59]
@andrei-tatar I just saw that Google has opened up the local SDK - https://developers.googleblog.com/2020/04/local-home-sdk-ready-for-actions.html?m=1

This is way over my paygrade. Any thoughts on how hard it would be to include a local fulfillment path in the current version of NORA or create another version for local use?

**Comment by andrei-tatar** [comment]
Already started on it a while back: https://github.com/andrei-tatar/nora-service/commit/e9798646f151fda2dd7c2b8280e269b44b504215
Just need to extract the command handling stuff from the service in a common package so it can be reused both in the service and in the node-red plugin.
It will probably take some work as it needs to remain backward compatible. Especially as local execution can fallback to cloud execution. So both need to be supported anyways.

**Comment by rgerrans** [comment]
Looking forward to it!!! Let me know if you need any help testing

**Comment by realjax** [comment]
> It will probably take some work as it needs to remain backward compatible. Especially as local execution can fallback to cloud execution. So both need to be supported anyways.

Does the fallback to cloud execution mean the noras service has to support the device as well for local execution? If so, could you make fallback to cloud execution optional maybe?  Maybe that way we could be less dependant on the number of device that nora's service support ?

**Comment by andrei-tatar** [comment]
> Does the fallback to cloud execution mean the noras service has to support the device as well for local execution? 

Yes. As far as I understood last time I read the plan at Google, local execution can be skipped at any time. Also, the queries for the state never go through the local execution so they always need to be in sync.


> If so, could you make fallback to cloud execution optional maybe? Maybe that way we could be less dependant on the number of device that nora's service support ?

Not possible right now. But maybe we could add some kind of a generic device where you can choose the type and the traits from a list of supported ones.
